Phobos-developers/Phobos
# CrateType extension

CrateType extension that allow user to attach Weapon and SuperWeapon to Crate goodies (Granted &  auto shot to collected cell)
How To Test : 
1.Go to : 
```
[Powerups] 
;Darkness chance must be > 0 in order to use the extension
Darkness=100,SHROUDX,yes              ; cloak entire radar map
```

2.RulesMD.ini 
```
[CrateTypes]
0=CRATENAME
```
```
[CRATENAME]
;it is better to use a clone SW for crate so it wont stack and interference with other bulding SW
;for example , [NUKESPECIAL] copy it and make [NUKESPECIAL_CRATE]
Crate.SuperWeapon= ;SuperWeaponType (Type 1) 
Crate.Weapon= ;WeaponType (Type 2)
Crate.SuperWeaponGrant= ;bool (single shot SW *Type1*)
Crate.Anim= ;Anim played when obtained
Crate.Chance= ;int (Appear chance)
Crate.Type = ;int (crate Type : 1-SuperWeapon Laucher , 2-WeaponLaucher , 3-Darkness (Re-shroud) , 4-Random Unit , 5-Money , 6-HealAll(test)
Crate.Sound= ;sound played when obtain
Crate.AllowWater= ;bool (allow to obtain on water if no , give money Rules->SoloCrateMoney)
Crate.EVA= ;EVA play when collected
Crate.MoneyMin ;int MoneyMin for Type 5 Crate
Crate.MoneyMax ;int  MoneyMax for Type 5 Crate
Crate.Units ;List Of Unit to Spawn with Type 4 Crate
```
//==============================================================
Unit with **CarriesCrate=yes** can spawn spesific crate defined by user with : 
```
[SOMEUNIT]
CarriesCrate=yes
CrateType= ;int (0-18 vanilla crate , 20 random crate , 20+ from CrateTypes list)
                 ;for example CrateType=21 mean will spawn "CRATENAME" from CrateTypes list
```

*When Spawning spesific crate Chance= ;int is ignored 
*some limitation  check were disabled atm for testing purposes  , will brought back later
//==============================================================
Building with **CrateBeneath=yes** can spawn spesific crate defined by user with : 
```
[SOMEBUILDING]
CrateBeneath=yes
CrateType= ;int (0-18 vanilla crate , 20 random crate , 20+ from CrateTypes list)
                 ;for example CrateType=21 mean will spawn "CRATENAME" from CrateTypes list
```

*When Spawning spesific crate Chance= ;int is ignored 
*some limitation  check were disabled atm for testing purposes  , will brought back later
//==============================================================
Spawn Crate thru Action Trigger 
User now can spawn additional crates thru **Trigger Action 108** 
**with CrateType= ;int (0-18 vanilla crate , 20 random crate , 20+ from CrateTypes list )**

 ;for example CrateType=21 mean will spawn "CRATENAME" from CrateTypes list
*When Spawning spesific crate Chance= ;int is ignored 
//=============================================================
Known Issues : 
-Some SW tag from Ares cant be controlled when using auto shot SW (type 1 , Crate.SWGrant=false)
-Droppod and UnitDelivery SW(s) will fail to place a unit/building when cell around crate is occupied
-Cant Use/interact/Fire with EMPulse SW(s) without EMPulse building present
-Online Multiplayer untested 
-Crate cannot be placed if there is creaters , tiberium , or any object that occupy the cell (WWP)
- 0-ing chance all Vanilla crate will cause Game to Freeze or IE (WWP)
